### PR TITLE
Fix/unknown Android classes

### PR DESCRIPTION
**Pull request by aSemy**
Hopefully will improve #71 

- [x] Set up Android integration test
- [x] Try triggering Android artifact transformations (as discovered in https://github.com/adamko-dev/dokkatoo/issues/71#issuecomment-1555058332)
- [x] tidy up code in Android/Kotlin adapters
- [x] improve setting up of Dokka test projects (so `local.properties` is generated properly)
- [x] Fix `Failed to install the following Android SDK packages as some licences have not been accepted.` error on GitHub action

#### Significant changes

* `DokkaSourceSetIdSpec.sourceSetName` mutability has been deprecated. Instances of this class is used in Sets, and so its hashcode/equals must remain constant.
* Minimum Android version updated to 4.2.2

### CODE REVIEW

**Comment by aSemy** [comment]
hey @EdricChan03 👋  Are you able to check out this branch, publish to to Maven Local, and give it a go in your project? I think it should fix your issue.

(the tests are failing because of a inconsequential difference in the generated HTML)

**Comment by EdricChan03** [comment]
It seems that the `index.html` isn't correctly generated? (Or rather, it appears to be in the wrong directory) It fails with the following exception:

```
Execution failed for task ':dokkatooGeneratePublicationHtml'.
> A failure occurred while executing dev.adamko.dokkatoo.workers.DokkaGeneratorWorker
   > /<...>/androidx-ktx-extras/build/dokkatoo/html/androidx/browser/browser-ktx/index.html
```

This is what the build directory structure looks like for that particular module:

Build directory structure for browser-ktx

I don't see any <ERROR class> thingy in the HTML output though, so I guess that's a partial success 🥳

**Comment by aSemy** [comment]
Thanks a lot for checking! It's a good sign.

I'm not sure what the problem might be with the index.html file. Possibly it's a bug from another change I've made since v1.3.0. 

I'll finish up the work in this PR and if there's another bug to fix, then we can make a new issue for it.

File: modules/dokkatoo-plugin/src/main/kotlin/adapters/DokkatooKotlinAdapter.kt [hunk 1]
```diff
@@ -278,7 +277,7 @@ private class KotlinCompilationDetailsBuilder(
       configurations.collectIncomingFiles(named = named, collector = compilationClasspath) {
         withVariantReselection()
         attributes {
-          attribute(USAGE_ATTRIBUTE, objects.named(JAVA_RUNTIME))
+          attribute(USAGE_ATTRIBUTE, objects.named(JAVA_API))
```
Comment (aSemy) [inline modules/dokkatoo-plugin/src/main/kotlin/adapters/DokkatooKotlinAdapter.kt:280]: I don't know if changing `JAVA_RUNTIME` to `JAVA_API` is significant 🤷‍♀️

**Pull request merged**